### ISSUE TITLE
Add command socket

### ISSUE THREAD

**Issue by kostaskol** [Infobserve/processor-rs#17]
From @leOug's AWESOME idea, the processor should be a server listening to a (unix) socket for commands (such as `start`, `stop`, `recompile` etc.).
This can be a bit tricky to do, as we need to explicitly have a controller thread that listens to the socket and distributes commands as needed.
Still, once implemented, this will lead us to a new (and quite exciting) direction

---

To implement this, we have to split it into smaller tasks (i.e. create a command socket, re-org the main thread to also be the controlling one, etc).

---

Lastly, this is blocked on #6 as it's best to have a complete working prototype before beginning a logic overhaul such as this
